### ISSUE TITLE
Ability to generate contexts after components for them have been defined

### ISSUE THREAD

**Issue by christianvoigt** [sschmid/Entitas-CSharp#487]
I experimented with different Entitas.properties files and messed up by using a properties file with a different list of contexts defined. This made me realize that it is currently not possible to generate a missing Context class, if Components for this context have already been defined. The following steps should reproduce the error:

1. Let's say in the first properties file Entitas.CodeGeneration.Plugins.Contexts contains "Context1, Context2". I generate these contexts and create components for them.
2. Now I switch to the second Entitas.properties file. This time Entitas.CodeGeneration.Plugins.Contexts contains only Context1. I run the generator, so that the generated files for Context2 are deleted.
3. Now I switch back to the first properties file and try to generate code using the Roslyn generator. The Roslyn generator will throw the following error:

```
System.NullReferenceException: Object reference not set to an instance of an object
  at Entitas.Roslyn.CodeGeneration.Plugins.ContextsComponentDataProvider.GetContextNames (Microsoft.CodeAnalysis.INamedTypeSymbol type) [0x0008b] in <25b7f79d3dd54b1e9af6b975b0f42a77>:0
  at Entitas.Roslyn.CodeGeneration.Plugins.ContextsComponentDataProvider.GetContextNamesOrDefault (Microsoft.CodeAnalysis.INamedTypeSymbol type) [0x00000] in <25b7f79d3dd54b1e9af6b975b0f42a77>:0
```
So I guess the generator comes across my components for Context2, collects the contextNames defined for these components, and then tries to find the missing class Context2 that has been deleted in step 2, right? 

I fixed this, by copying the Sources/Generated folder from a backup that contained all Context classes. But I think this issue should be handled by the generator, as the error in the CLI is not self-explanatory for everyone (if one does run without -v it is even more obscure).

Ideally the generator would automatically regenerate missing Context classes before collecting the ComponentData, so that it would even be possible to delete the entire Sources/Generated folder and regenerate everything at once.

**Comment by sschmid** [comment]
Good find! Changing the priority of the roslyn plugins to run after the context won't help, because it depends on the actual code, meaning it needs to generate first. I will see if I can find a solution

**Comment by sschmid** [comment]
Fixed in 0.47.x
